### PR TITLE
MAGENTO-18131: Fixed EAV attributes values query

### PR DESCRIPTION
**Pull request by strell**
### Description
SQL Join error when using the custom EAV entity together with the standard eav_entity entity table.
Had to add the table alias into the entity_type_id filter condition.

### Fixed Issues (if relevant)
1. magento/magento2#18131: Entity Type ID at Join 

### Manual testing scenarios
See issue description

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18437) <br/>All committers have signed the CLA.

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-3124](https://jira.corp.magento.com/browse/ENGCOM-3124) has been created to process this Pull Request

**Comment by aleron75** [comment]
Hello @strell, thanks for your contribution.
Once merged, don't forget you can easily port this PR to other branches, if applicable, with the Porting Tool, [read more here](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically).

**Comment by magento-engcom-team** [comment]
Hi @strell. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**
